### DISCUSSION TITLE
Why not just write it as wfexec.NewSession(execCtx, s.graph, opts...)

### DISCUSSION THREAD

**Discussion by showntop** [cortezaproject/corteza-server#418] [Q&A] [answered]
https://github.com/cortezaproject/corteza-server/blob/57721291b270d4ed9b9e625d7a17fca2cc6c27eb/automation/service/session.go#L322

I'm confused why use channel block is't running. I think it's same as 
`ses = types.NewSession(wfexec.NewSession(execCtx, s.graph, opts...))
`
why not just write it like this simply.

**Comment by darh** [comment, marked as answer]
not quite clear what the question is :)

We need to create wf session in the background, detached from the current (request) context and "bind it" to the main app context. 
Main reason is having more control over when session is canceled.
First, we do not want to be a victim of prematurely canceled HTTP request that invoked the session. Second we want to terminate sessions on server shutdown. 

Hope this helps.

I'm sure there are other approaches that address this particular challange.

If you have a suggestion how to rewrite the explanation above `spawn()` fn I'm happy to replace it.
